### PR TITLE
bpo-34129: Add private variables mangling in cgitb.py

### PR DESCRIPTION
**Pull request by pjurkas**
https://bugs.python.org/issue34129

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Unfortunately our records indicate you have not signed the CLA. For legal reasons we need you to sign this before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.


When your account is ready, please add a comment in this pull request
and a Python core developer will remove the `CLA not signed` label
to make the bot check again.

Thanks again for your contribution, we look forward to reviewing it!

**Comment by serhiy-storchaka** [comment]
Would be nice to add a test.

**Comment by pjurkas** [comment]
Unit test has been added. Thank you!

**Review by remilapeyre — APPROVED** [review]

**Comment by remilapeyre** [comment]
Thanks @pjurkas!

File: Lib/cgitb.py [hunk 1]
```diff
@@ -85,7 +85,10 @@ def scanvars(reader, frame, locals):
         if ttype == tokenize.NAME and token not in keyword.kwlist:
             if lasttoken == '.':
                 if parent is not __UNDEF__:
-                    value = getattr(parent, token, __UNDEF__)
+                    mangled = '_{0}{1}'.format(
+                        parent.__class__.__name__, token
+                        ) if token.startswith('__') else token
+                    value = getattr(parent, mangled, __UNDEF__)
```
Comment (benjaminp) [inline Lib/cgitb.py:91]: Does this work properly if `parent.__class__` is a subclass the of the lexically enclosing class?

**Comment by csabella** [comment]
@pjurkas, please respond to the code review question.  Thank you!

**Pull request closed without merging**